### PR TITLE
Data structures assignment and notes

### PR DESCRIPTION
**Pull request by asmockler**
Improve directory structure for data structures section of the class. Adds assignments and examples.

We will need to update learning suite to reflect these changes after merge.

Sorry for the huge PR. Easiest to just [browse the changes](https://github.com/byu-is-403/syllabus/tree/data-structures/data-structures).

cc: @danmo91 @profganderson

### CODE REVIEW

**Comment by asmockler** [comment]
Ready to merge this and update LearningSuite once I get a 👍 :)

File: data-structures/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Data Structures
+
```
Comment (danmo91) [inline data-structures/README.md:2]: 👍 structure looks clean af

File: data-structures/assignment/data-structures-basics.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Data Structures Basics Assignment
+
+Your task is to create a program that will help you familiarize yourself with various data structures available in the C# language.
+
+You are the owner of a very successful hamburger restaurant. Your faithful customers line up every day and eat dozens of burgers. You are writing a program to track exactly how many hamburgers each customer eats.
```
Comment (danmo91) [inline data-structures/assignment/data-structures-basics.md:5]: 🍔

**Review by danmo91 — APPROVED** [review]
Nice job @asmockler !

Do you need any help putting this on Learning Suite while you are out of town??

**Pull request merged**